packages/react-composites/tests/browser/common/utils.ts

```diff
@@ -2,33 +2,76 @@
 // Licensed under the MIT license.
 
 import { IDS } from './constants';
-import { Page } from '@playwright/test';
+import { ElementHandle, JSHandle, Page } from '@playwright/test';
 import { ChatUserType, CallUserType, MeetingUserType } from './fixtureTypes';
 
+// This timeout must be less than the global timeout
+export const PER_STEP_TIMEOUT_MS = 3000;
+
 /** Selector string to get element by data-ui-id property */
 export const dataUiId = (id: string): string => `[data-ui-id="${id}"]`;
 
+async function screenshotOnFailure<T>(page: Page, fn: () => Promise<T>): Promise<T> {
+  try {
+    return await fn();
+  } catch (e) {
+    await page.screenshot({ path: 'test-results/failure-screenshot.png' });
```

Comment:
I'd recommend adding a random suffix here. That way if we do somehow end up hitting this more than once in CI, we won't overwrite files.

Reply:
Good note, updated to `test-results/failure-screenshot-${generateGUID()}.png`

Reply:
Actually, turned on `screenshot: 'only-on-failure'` in the settings, which now works because of the timeout

Reply:
OK noo reverted that, it gives weird results on mismatches:
![image](https://user-images.githubusercontent.com/2684369/143285452-4a603d15-dd12-48b5-8901-b6532eb0d0d6.png)

I don't know when these `only-on-failure` are actually taken but none are at the end....
| expected | actual | test-failed-1.png | test-failed-2.png |
| -- | -- | -- | -- |
|![image](https://user-images.githubusercontent.com/2684369/143285406-b4fdfb2c-02d1-4c85-b5fa-d58d0c00e882.png)|![image](https://user-images.githubusercontent.com/2684369/143285391-08e31b74-c869-4246-b0b7-02950c27a453.png)|![image](https://user-images.githubusercontent.com/2684369/143285420-2c576c10-f6ed-4217-a72d-2d866bb25c3f.png)|![image](https://user-images.githubusercontent.com/2684369/143285463-320287a1-6ac6-47ff-82c4-f01d349135ad.png)|

Reply:
reverted back to custom screenshot method (with guid for uniqueness)